<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Create Enrichment map from R with GSEA results | Translate gmt file from human to another species</title>
  <meta name="description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Create Enrichment map from R with GSEA results | Translate gmt file from human to another species" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/cover.png" />
  <meta property="og:description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="github-repo" content="risserlin/CBW_pathways_workshop_R_notebooks" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Create Enrichment map from R with GSEA results | Translate gmt file from human to another species" />
  
  <meta name="twitter:description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="twitter:image" content="/images/cover.png" />

<meta name="author" content="Ruth Isserlin" />


<meta name="date" content="2024-06-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="create-enrichment-map-from-r-with-gprofiler-results.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Workshop R Notebooks</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Index</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> CBW Workshop example R Notebooks</a></li>
<li class="chapter" data-level="3" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="setup.html"><a href="setup.html#install-r-and-rstudio"><i class="fa fa-check"></i><b>3.1</b> Install R and RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="setup.html"><a href="setup.html#docker-optional"><i class="fa fa-check"></i><b>3.2</b> Docker [Optional]</a></li>
<li class="chapter" data-level="3.3" data-path="setup.html"><a href="setup.html#install-docker"><i class="fa fa-check"></i><b>3.3</b> Install Docker</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="setup.html"><a href="setup.html#windows"><i class="fa fa-check"></i><b>3.3.1</b> Windows</a></li>
<li class="chapter" data-level="3.3.2" data-path="setup.html"><a href="setup.html#macos-linux"><i class="fa fa-check"></i><b>3.3.2</b> MacOS / Linux</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html"><i class="fa fa-check"></i><b>4</b> Run g:profiler from R</a>
<ul>
<li class="chapter" data-level="4.1" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#initialize-variables-and-libraries"><i class="fa fa-check"></i><b>4.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="4.2" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#load-in-query-set"><i class="fa fa-check"></i><b>4.2</b> Load in Query set</a></li>
<li class="chapter" data-level="4.3" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#run-gprofiler-with-supplied-genesets"><i class="fa fa-check"></i><b>4.3</b> Run g:profiler with supplied genesets</a></li>
<li class="chapter" data-level="4.4" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#download-and-load-gprofiler-geneset-file"><i class="fa fa-check"></i><b>4.4</b> Download and load g:profiler geneset file</a></li>
<li class="chapter" data-level="4.5" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#filter-results-by-geneset-size"><i class="fa fa-check"></i><b>4.5</b> Filter results by geneset size</a></li>
<li class="chapter" data-level="4.6" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#create-an-output-file-of-the-results---generic-enrichment-map-file-from-gprofiler-gmt"><i class="fa fa-check"></i><b>4.6</b> Create an output file of the results - Generic enrichment Map file from g:profiler gmt</a></li>
<li class="chapter" data-level="4.7" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#run-gprofiler-with-your-own-genesets-example-using-baderlab-genesets"><i class="fa fa-check"></i><b>4.7</b> Run g:profiler with your own genesets (example using BaderLab genesets)</a></li>
<li class="chapter" data-level="4.8" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#download-and-load-bader-lab-geneset-file"><i class="fa fa-check"></i><b>4.8</b> Download and load Bader lab geneset file</a></li>
<li class="chapter" data-level="4.9" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#filter-bader-lab-geneset-file"><i class="fa fa-check"></i><b>4.9</b> Filter Bader lab geneset file</a></li>
<li class="chapter" data-level="4.10" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#upload-the-gmt-files-to-gprofiler"><i class="fa fa-check"></i><b>4.10</b> Upload the gmt files to gprofiler</a></li>
<li class="chapter" data-level="4.11" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#create-an-output-file-of-the-results---generic-enrichment-map-file-from-baderlab-gmt"><i class="fa fa-check"></i><b>4.11</b> Create an output file of the results - Generic enrichment Map file from Baderlab gmt</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html"><i class="fa fa-check"></i><b>5</b> Run GSEA from within R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#load-in-required-libraries"><i class="fa fa-check"></i><b>5.1</b> Load in required libraries</a></li>
<li class="chapter" data-level="5.2" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#configurable-parameters"><i class="fa fa-check"></i><b>5.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#download-the-latest-pathway-definition-file"><i class="fa fa-check"></i><b>5.3</b> Download the latest pathway definition file</a></li>
<li class="chapter" data-level="5.4" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#run-gsea"><i class="fa fa-check"></i><b>5.4</b> Run GSEA</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html"><i class="fa fa-check"></i><b>6</b> Create GMT file from Ensembl</a>
<ul>
<li class="chapter" data-level="6.1" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#load-libraries"><i class="fa fa-check"></i><b>6.1</b> Load Libraries</a></li>
<li class="chapter" data-level="6.2" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#set-up-biomart-connection"><i class="fa fa-check"></i><b>6.2</b> Set up Biomart connection</a></li>
<li class="chapter" data-level="6.3" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#get-species-go-annotations"><i class="fa fa-check"></i><b>6.3</b> Get species GO annotations</a></li>
<li class="chapter" data-level="6.4" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#format-results-into-gmt-file"><i class="fa fa-check"></i><b>6.4</b> Format results into GMT file</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html"><i class="fa fa-check"></i><b>7</b> Translate human gmt file to another species using ensembl</a>
<ul>
<li class="chapter" data-level="7.1" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#load-in-required-libraries-1"><i class="fa fa-check"></i><b>7.1</b> Load in required libraries</a></li>
<li class="chapter" data-level="7.2" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#download-the-latest-pathway-definition-file-1"><i class="fa fa-check"></i><b>7.2</b> Download the latest pathway definition file</a></li>
<li class="chapter" data-level="7.3" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#set-up-biomart-connection-1"><i class="fa fa-check"></i><b>7.3</b> Set up Biomart connection</a></li>
<li class="chapter" data-level="7.4" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#convert-the-species-genes-from-human"><i class="fa fa-check"></i><b>7.4</b> Convert the species genes from human</a></li>
<li class="chapter" data-level="7.5" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#format-results-into-gmt-file-1"><i class="fa fa-check"></i><b>7.5</b> Format results into GMT file</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html"><i class="fa fa-check"></i><b>8</b> Create Enrichment map from R with g:Profiler results</a>
<ul>
<li class="chapter" data-level="8.1" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#initialize-variables-and-libraries-1"><i class="fa fa-check"></i><b>8.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="8.2" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#configurable-parameters-1"><i class="fa fa-check"></i><b>8.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="8.3" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#specify-data-files"><i class="fa fa-check"></i><b>8.3</b> Specify Data files</a></li>
<li class="chapter" data-level="8.4" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#launch-cytoscape"><i class="fa fa-check"></i><b>8.4</b> Launch Cytoscape</a></li>
<li class="chapter" data-level="8.5" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#make-sure-you-can-connect-to-cytoscape"><i class="fa fa-check"></i><b>8.5</b> Make sure you can connect to Cytoscape</a></li>
<li class="chapter" data-level="8.6" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#create-an-enrichment-map"><i class="fa fa-check"></i><b>8.6</b> Create an Enrichment map</a></li>
<li class="chapter" data-level="8.7" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#create-an-enrichment-map---run-em-command"><i class="fa fa-check"></i><b>8.7</b> Create an Enrichment map - run EM command</a></li>
<li class="chapter" data-level="8.8" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#get-a-screen-shot-of-the-initial-network."><i class="fa fa-check"></i><b>8.8</b> Get a screen shot of the initial network.</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html"><i class="fa fa-check"></i><b>9</b> Create Enrichment map from R with GSEA results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#initialize-variables-and-libraries-2"><i class="fa fa-check"></i><b>9.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="9.2" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#configurable-parameters-2"><i class="fa fa-check"></i><b>9.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="9.3" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#specify-data-files-1"><i class="fa fa-check"></i><b>9.3</b> Specify Data files</a></li>
<li class="chapter" data-level="9.4" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#optional-file-specification"><i class="fa fa-check"></i><b>9.4</b> Optional File specification</a></li>
<li class="chapter" data-level="9.5" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#launch-cytoscape-1"><i class="fa fa-check"></i><b>9.5</b> Launch Cytoscape</a></li>
<li class="chapter" data-level="9.6" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#make-sure-you-can-connect-to-cytoscape-1"><i class="fa fa-check"></i><b>9.6</b> Make sure you can connect to Cytoscape</a></li>
<li class="chapter" data-level="9.7" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map-1"><i class="fa fa-check"></i><b>9.7</b> Create an Enrichment map</a></li>
<li class="chapter" data-level="9.8" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map---run-em-command-1"><i class="fa fa-check"></i><b>9.8</b> Create an Enrichment map - run EM command</a></li>
<li class="chapter" data-level="9.9" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#get-a-screen-shot-of-the-initial-network.-1"><i class="fa fa-check"></i><b>9.9</b> Get a screen shot of the initial network.</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>10</b> References</a></li>
<li class="divider"></li>
<li><a href="https://baderlab.org" target="blank"><img src="./images/baderlablogo.png" width="280">Bader Lab</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Translate gmt file from human to another species</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="create-enrichment-map-from-r-with-gsea-results" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Create Enrichment map from R with GSEA results<a href="create-enrichment-map-from-r-with-gsea-results.html#create-enrichment-map-from-r-with-gsea-results" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="initialize-variables-and-libraries-2" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Initialize variables and libraries<a href="create-enrichment-map-from-r-with-gsea-results.html#initialize-variables-and-libraries-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-1" tabindex="-1"></a><span class="co">#use library</span></span>
<span id="cb87-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-2" tabindex="-1"></a><span class="co">#make sure biocManager is installed</span></span>
<span id="cb87-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-3" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;BiocManager&quot;</span>)}, </span>
<span id="cb87-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-4" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) { </span>
<span id="cb87-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-5" tabindex="-1"></a>           <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)}, </span>
<span id="cb87-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-6" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;BiocManager&quot;</span>))</span>
<span id="cb87-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-7" tabindex="-1"></a></span>
<span id="cb87-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-8" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)}, </span>
<span id="cb87-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-9" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) { <span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)}, </span>
<span id="cb87-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-10" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>))</span>
<span id="cb87-11"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-11" tabindex="-1"></a></span>
<span id="cb87-12"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-12" tabindex="-1"></a><span class="co">#use easy cyRest library to communicate with cytoscape.</span></span>
<span id="cb87-13"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-13" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;RCy3&quot;</span>)}, </span>
<span id="cb87-14"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-14" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) { BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;RCy3&quot;</span>)}, </span>
<span id="cb87-15"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-15" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;RCy3&quot;</span>))</span>
<span id="cb87-16"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-16" tabindex="-1"></a></span>
<span id="cb87-17"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-17" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;httr&quot;</span>)}, </span>
<span id="cb87-18"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-18" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) { BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;httr&quot;</span>)}, </span>
<span id="cb87-19"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb87-19" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;httr&quot;</span>))</span></code></pre></div>
</div>
<div id="configurable-parameters-2" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Configurable Parameters<a href="create-enrichment-map-from-r-with-gsea-results.html#configurable-parameters-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-1" tabindex="-1"></a><span class="co"># is_docker - true/false depending on if you are running R from docker</span></span>
<span id="cb88-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-2" tabindex="-1"></a>is_docker <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb88-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-3" tabindex="-1"></a></span>
<span id="cb88-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-4" tabindex="-1"></a><span class="co">#directory where all the original input data file are</span></span>
<span id="cb88-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-5" tabindex="-1"></a><span class="co"># for example ./data/</span></span>
<span id="cb88-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-6" tabindex="-1"></a>working_dir <span class="ot">&lt;-</span> params<span class="sc">$</span>working_dir</span>
<span id="cb88-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-7" tabindex="-1"></a></span>
<span id="cb88-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-8" tabindex="-1"></a></span>
<span id="cb88-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-9" tabindex="-1"></a><span class="co">#directory where all the generated data files are found.</span></span>
<span id="cb88-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-10" tabindex="-1"></a><span class="co"># For example - ./generated_data/</span></span>
<span id="cb88-11"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-11" tabindex="-1"></a><span class="co"># If you are using all the notebooks from this set the generated data will be</span></span>
<span id="cb88-12"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-12" tabindex="-1"></a><span class="co"># put in the ./generated_data folder.  You have to specify if it is gsea or </span></span>
<span id="cb88-13"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-13" tabindex="-1"></a><span class="co"># gprofiler</span></span>
<span id="cb88-14"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-14" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> params<span class="sc">$</span>output_dir</span>
<span id="cb88-15"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-15" tabindex="-1"></a></span>
<span id="cb88-16"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-16" tabindex="-1"></a></span>
<span id="cb88-17"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-17" tabindex="-1"></a><span class="co">#defined threshold for GSEA enrichments </span></span>
<span id="cb88-18"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-18" tabindex="-1"></a><span class="co">#p-value to filter all the genesets.  For example -   1.0</span></span>
<span id="cb88-19"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-19" tabindex="-1"></a>pvalue_gsea_threshold <span class="ot">&lt;-</span> params<span class="sc">$</span>pvalue_thresh</span>
<span id="cb88-20"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-20" tabindex="-1"></a></span>
<span id="cb88-21"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-21" tabindex="-1"></a><span class="co">#q-value to filter all the genesets.  For example -   0.05</span></span>
<span id="cb88-22"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-22" tabindex="-1"></a>qvalue_gsea_threshold <span class="ot">&lt;-</span> params<span class="sc">$</span>qvalue_thresh</span>
<span id="cb88-23"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-23" tabindex="-1"></a></span>
<span id="cb88-24"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-24" tabindex="-1"></a><span class="co">#similarity threshold to filter all the genesets connections/edges.  </span></span>
<span id="cb88-25"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-25" tabindex="-1"></a><span class="co"># For example -   0.375</span></span>
<span id="cb88-26"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-26" tabindex="-1"></a>similarity_threshold <span class="ot">&lt;-</span> <span class="st">&quot;0.375&quot;</span></span>
<span id="cb88-27"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-27" tabindex="-1"></a></span>
<span id="cb88-28"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-28" tabindex="-1"></a><span class="co">#similarity metric to filter all the genesets connections/edges </span></span>
<span id="cb88-29"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-29" tabindex="-1"></a><span class="co"># (can be OVERLAP, JACCARD, or COMBINED.   For example -   Combined</span></span>
<span id="cb88-30"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb88-30" tabindex="-1"></a>similarity_metric <span class="ot">=</span> <span class="st">&quot;COMBINED&quot;</span></span></code></pre></div>
</div>
<div id="specify-data-files-1" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Specify Data files<a href="create-enrichment-map-from-r-with-gsea-results.html#specify-data-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Depending on whether you are creating your enrichment map from g:Profiler or GSEA results the sets of files might be a little different. Minimally, you will need to specify:
* gmt file
* enrichment results file</p>
<p>Although there is a gmt file in the gsea edb results directory(which is the easiest method to create an enrichment map) it have been filtered to contain only genes represented in the expression set. If you use this fltered file you will get different pathway connectivity depending on the dataset being used. We recommend using original gmt file used for the gsea analysis and not the filtered one in the results directory.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-1" tabindex="-1"></a><span class="co">#use the newest gmt file in the output directory</span></span>
<span id="cb89-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-2" tabindex="-1"></a>gmt_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> output_dir, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.gmt&quot;</span>)</span>
<span id="cb89-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-3" tabindex="-1"></a></span>
<span id="cb89-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-4" tabindex="-1"></a>  <span class="co">#get the details on the files</span></span>
<span id="cb89-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-5" tabindex="-1"></a>  details <span class="ot">=</span> <span class="fu">file.info</span>(<span class="fu">file.path</span>(output_dir,gmt_files))</span>
<span id="cb89-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-6" tabindex="-1"></a>  <span class="co">#order according to newest to oldest</span></span>
<span id="cb89-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-7" tabindex="-1"></a>  details <span class="ot">=</span> details[<span class="fu">with</span>(details, <span class="fu">order</span>(<span class="fu">as.POSIXct</span>(mtime),<span class="at">decreasing =</span> <span class="cn">TRUE</span>)), ]</span>
<span id="cb89-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-8" tabindex="-1"></a></span>
<span id="cb89-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-9" tabindex="-1"></a>  <span class="co">#use the newest file:</span></span>
<span id="cb89-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb89-10" tabindex="-1"></a> gmt_gsea_file <span class="ot">&lt;-</span> <span class="fu">row.names</span>(details)[<span class="dv">1</span>]</span></code></pre></div>
<p>GSEA output directory - You can specify the exact name of the directory. The below code looks for the newest GSEA results directory and uses that.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-1" tabindex="-1"></a>gsea_directories <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> output_dir, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.GseaPreranked&quot;</span>)</span>
<span id="cb90-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-2" tabindex="-1"></a></span>
<span id="cb90-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-3" tabindex="-1"></a><span class="co">#get the details on the files</span></span>
<span id="cb90-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-4" tabindex="-1"></a>details <span class="ot">=</span> <span class="fu">file.info</span>(<span class="fu">file.path</span>(output_dir,gsea_directories))</span>
<span id="cb90-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-5" tabindex="-1"></a><span class="co">#order according to newest to oldest</span></span>
<span id="cb90-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-6" tabindex="-1"></a>details <span class="ot">=</span> details[<span class="fu">with</span>(details, <span class="fu">order</span>(<span class="fu">as.POSIXct</span>(mtime),<span class="at">decreasing =</span> <span class="cn">TRUE</span>)), ]</span>
<span id="cb90-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-7" tabindex="-1"></a></span>
<span id="cb90-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-8" tabindex="-1"></a><span class="co">#use the newest file:</span></span>
<span id="cb90-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-9" tabindex="-1"></a>gsea_output_dir <span class="ot">&lt;-</span> <span class="fu">row.names</span>(details)[<span class="dv">1</span>]</span>
<span id="cb90-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-10" tabindex="-1"></a></span>
<span id="cb90-11"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-11" tabindex="-1"></a>gsea_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(gsea_output_dir,<span class="st">&quot;edb&quot;</span>)</span>
<span id="cb90-12"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb90-12" tabindex="-1"></a>gsea_results_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(gsea_results_path,<span class="st">&quot;results.edb&quot;</span>)</span></code></pre></div>
</div>
<div id="optional-file-specification" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Optional File specification<a href="create-enrichment-map-from-r-with-gsea-results.html#optional-file-specification" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>These files are not needed to create the enrichment map but are very beneficial when analyzing your result.<br />
* gene expression file
* gene ranks file</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-1" tabindex="-1"></a>gsea_ranks_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(gsea_results_path,</span>
<span id="cb91-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-2" tabindex="-1"></a>                             <span class="fu">list.files</span>(gsea_results_path,<span class="at">pattern=</span><span class="st">&quot;.rnk&quot;</span>))</span>
<span id="cb91-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-3" tabindex="-1"></a></span>
<span id="cb91-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-4" tabindex="-1"></a>expression_file_fullpath <span class="ot">&lt;-</span> <span class="fu">file.path</span>(working_dir,</span>
<span id="cb91-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-5" tabindex="-1"></a>                          params<span class="sc">$</span>expression_file)</span>
<span id="cb91-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-6" tabindex="-1"></a></span>
<span id="cb91-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-7" tabindex="-1"></a>cls_file_fullpath <span class="ot">&lt;-</span> <span class="fu">file.path</span>(working_dir, params<span class="sc">$</span>cls_file)</span>
<span id="cb91-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-8" tabindex="-1"></a></span>
<span id="cb91-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-9" tabindex="-1"></a><span class="co">#define an analysis name</span></span>
<span id="cb91-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb91-10" tabindex="-1"></a>cur_model_name <span class="ot">&lt;-</span> params<span class="sc">$</span>analysis_name</span></code></pre></div>
</div>
<div id="launch-cytoscape-1" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Launch Cytoscape<a href="create-enrichment-map-from-r-with-gsea-results.html#launch-cytoscape-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Launch Cytoscape (by default cytoscape will automatically enable rest so as long as cytoscape 3.3 or higher is open R should be able to communicate with it). Make sure if you get an message asking you if you want communicate with other apps that you select “Allow”.</p>
</div>
<div id="make-sure-you-can-connect-to-cytoscape-1" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Make sure you can connect to Cytoscape<a href="create-enrichment-map-from-r-with-gsea-results.html#make-sure-you-can-connect-to-cytoscape-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-1" tabindex="-1"></a><span class="cf">if</span>(is_docker){</span>
<span id="cb92-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-2" tabindex="-1"></a>  current_base <span class="ot">=</span> <span class="st">&quot;host.docker.internal:1234/v1&quot;</span></span>
<span id="cb92-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-3" tabindex="-1"></a>  .defaultBaseUrl <span class="ot">&lt;-</span> <span class="st">&quot;http://host.docker.internal:1234/v1&quot;</span></span>
<span id="cb92-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-4" tabindex="-1"></a>} <span class="cf">else</span>{</span>
<span id="cb92-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-5" tabindex="-1"></a>  current_base <span class="ot">=</span> <span class="st">&quot;localhost:1234/v1&quot;</span></span>
<span id="cb92-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-6" tabindex="-1"></a>}</span>
<span id="cb92-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-7" tabindex="-1"></a></span>
<span id="cb92-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb92-8" tabindex="-1"></a><span class="fu">cytoscapePing</span> (<span class="at">base.url =</span> current_base)</span></code></pre></div>
<pre><code>## You are connected to Cytoscape!</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb94-1" tabindex="-1"></a><span class="fu">cytoscapeVersionInfo</span> (<span class="at">base.url =</span> current_base)</span></code></pre></div>
<pre><code>##       apiVersion cytoscapeVersion 
##             &quot;v1&quot;         &quot;3.10.2&quot;</code></pre>
<hr />
</div>
<div id="create-an-enrichment-map-1" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Create an Enrichment map<a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you are running R from within a docker you need to first upload your datafiles to Cytoscape before you can create your enrichment map</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-1" tabindex="-1"></a><span class="co">#if using docker we need to replace all the the paths to the host path</span></span>
<span id="cb96-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-2" tabindex="-1"></a><span class="cf">if</span>(is_docker) {</span>
<span id="cb96-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-3" tabindex="-1"></a>  upload_em_file <span class="ot">&lt;-</span> <span class="cf">function</span>(localPath) {</span>
<span id="cb96-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-4" tabindex="-1"></a>    bname <span class="ot">&lt;-</span> <span class="fu">basename</span>(localPath)</span>
<span id="cb96-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-5" tabindex="-1"></a>    r <span class="ot">&lt;-</span> <span class="fu">POST</span>(</span>
<span id="cb96-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-6" tabindex="-1"></a>      <span class="at">url =</span> </span>
<span id="cb96-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-7" tabindex="-1"></a><span class="fu">paste</span>(<span class="st">&#39;http://host.docker.internal:1234/enrichmentmap/textfileupload?fileName=&#39;</span>, </span>
<span id="cb96-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-8" tabindex="-1"></a>                  bname, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb96-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-9" tabindex="-1"></a>      <span class="at">config =</span> <span class="fu">list</span>(),</span>
<span id="cb96-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-10" tabindex="-1"></a>      <span class="at">body =</span> <span class="fu">list</span>(<span class="at">file =</span> <span class="fu">upload_file</span>(localPath)),</span>
<span id="cb96-11"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-11" tabindex="-1"></a>      <span class="at">encode =</span> <span class="st">&quot;multipart&quot;</span>,</span>
<span id="cb96-12"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-12" tabindex="-1"></a>      <span class="at">handle =</span> <span class="cn">NULL</span></span>
<span id="cb96-13"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-13" tabindex="-1"></a>    )</span>
<span id="cb96-14"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-14" tabindex="-1"></a>    <span class="fu">content</span>(r,<span class="st">&quot;parsed&quot;</span>)<span class="sc">$</span>path</span>
<span id="cb96-15"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-15" tabindex="-1"></a>  }</span>
<span id="cb96-16"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-16" tabindex="-1"></a>  </span>
<span id="cb96-17"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-17" tabindex="-1"></a>  <span class="co"># &quot;upload&quot; the files to the host machine and replace each path </span></span>
<span id="cb96-18"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-18" tabindex="-1"></a>  <span class="co"># with the host machine path</span></span>
<span id="cb96-19"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-19" tabindex="-1"></a>  expression_file_fullpath <span class="ot">&lt;-</span> <span class="fu">upload_em_file</span>(expression_file_fullpath)</span>
<span id="cb96-20"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-20" tabindex="-1"></a>  class_file_fullpath <span class="ot">&lt;-</span> <span class="fu">upload_em_file</span>(cls_file_fullpath)</span>
<span id="cb96-21"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-21" tabindex="-1"></a>  gmt_gsea_file <span class="ot">&lt;-</span> <span class="fu">upload_em_file</span>(gmt_gsea_file)</span>
<span id="cb96-22"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-22" tabindex="-1"></a>  gsea_ranks_file <span class="ot">&lt;-</span> <span class="fu">upload_em_file</span>(gsea_ranks_file)</span>
<span id="cb96-23"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-23" tabindex="-1"></a>  gsea_results_filename <span class="ot">&lt;-</span> <span class="fu">upload_em_file</span>(gsea_results_filename)</span>
<span id="cb96-24"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb96-24" tabindex="-1"></a>}</span></code></pre></div>
<hr />
</div>
<div id="create-an-enrichment-map---run-em-command-1" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> Create an Enrichment map - run EM command<a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map---run-em-command-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-1" tabindex="-1"></a><span class="do">#######################################</span></span>
<span id="cb97-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-2" tabindex="-1"></a><span class="co">#create EM</span></span>
<span id="cb97-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-3" tabindex="-1"></a>current_network_name <span class="ot">&lt;-</span> <span class="fu">paste</span>(cur_model_name,pvalue_gsea_threshold,</span>
<span id="cb97-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-4" tabindex="-1"></a>                              qvalue_gsea_threshold,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb97-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-5" tabindex="-1"></a></span>
<span id="cb97-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-6" tabindex="-1"></a>em_command <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&#39;enrichmentmap build analysisType=&quot;gsea&quot; gmtFile=&#39;</span>,</span>
<span id="cb97-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-7" tabindex="-1"></a>                                                              gmt_gsea_file,</span>
<span id="cb97-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-8" tabindex="-1"></a>                   <span class="st">&#39;pvalue=&#39;</span>,pvalue_gsea_threshold, </span>
<span id="cb97-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-9" tabindex="-1"></a>                   <span class="st">&#39;qvalue=&#39;</span>,qvalue_gsea_threshold,</span>
<span id="cb97-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-10" tabindex="-1"></a>                   <span class="st">&#39;similaritycutoff=&#39;</span>,similarity_threshold,</span>
<span id="cb97-11"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-11" tabindex="-1"></a>                   <span class="st">&#39;coefficients=&#39;</span>,similarity_metric,</span>
<span id="cb97-12"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-12" tabindex="-1"></a>                   <span class="st">&#39;ranksDataset1=&#39;</span>, gsea_ranks_file,</span>
<span id="cb97-13"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-13" tabindex="-1"></a>                   <span class="st">&#39;enrichmentsDataset1=&#39;</span>,gsea_results_filename, </span>
<span id="cb97-14"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-14" tabindex="-1"></a>                   <span class="st">&#39;filterByExpressions=false&#39;</span>,</span>
<span id="cb97-15"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-15" tabindex="-1"></a>                   <span class="st">&#39;expressionDataset1=&#39;</span>,expression_file_fullpath,</span>
<span id="cb97-16"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-16" tabindex="-1"></a>                   <span class="st">&#39;classDataset1=&#39;</span>,class_file_fullpath,</span>
<span id="cb97-17"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-17" tabindex="-1"></a>                   <span class="st">&#39;gmtFile=&#39;</span>,gmt_gsea_file,</span>
<span id="cb97-18"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-18" tabindex="-1"></a>                   <span class="at">sep=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb97-19"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-19" tabindex="-1"></a></span>
<span id="cb97-20"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-20" tabindex="-1"></a><span class="co">#enrichment map command will return the suid of newly created network.</span></span>
<span id="cb97-21"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-21" tabindex="-1"></a>response <span class="ot">&lt;-</span> <span class="fu">commandsGET</span>(em_command,<span class="at">base.url =</span> current_base)</span>
<span id="cb97-22"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-22" tabindex="-1"></a></span>
<span id="cb97-23"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-23" tabindex="-1"></a>current_network_suid <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb97-24"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-24" tabindex="-1"></a><span class="co">#enrichment map command will return the suid of newly created network </span></span>
<span id="cb97-25"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-25" tabindex="-1"></a><span class="co"># unless it Failed.  If it failed it will contain the word failed</span></span>
<span id="cb97-26"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-26" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">grepl</span>(<span class="at">pattern=</span><span class="st">&quot;Failed&quot;</span>, response)){</span>
<span id="cb97-27"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-27" tabindex="-1"></a>  <span class="fu">paste</span>(response)</span>
<span id="cb97-28"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-28" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb97-29"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-29" tabindex="-1"></a>  current_network_suid <span class="ot">&lt;-</span> response</span>
<span id="cb97-30"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-30" tabindex="-1"></a>}</span>
<span id="cb97-31"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-31" tabindex="-1"></a></span>
<span id="cb97-32"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-32" tabindex="-1"></a><span class="co">#check to see if the network name is unique</span></span>
<span id="cb97-33"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-33" tabindex="-1"></a>current_names <span class="ot">&lt;-</span> <span class="fu">getNetworkList</span>(<span class="at">base.url =</span> current_base)</span>
<span id="cb97-34"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-34" tabindex="-1"></a><span class="cf">if</span>(current_network_name <span class="sc">%in%</span> current_names){</span>
<span id="cb97-35"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-35" tabindex="-1"></a>  <span class="co">#if the name already exists in the network names then put the SUID in front</span></span>
<span id="cb97-36"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-36" tabindex="-1"></a>  <span class="co"># of the name (this does not work if you put the suid at the end of the name)</span></span>
<span id="cb97-37"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-37" tabindex="-1"></a>  current_network_name <span class="ot">&lt;-</span> <span class="fu">paste</span>(current_network_suid,</span>
<span id="cb97-38"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-38" tabindex="-1"></a>                                current_network_name,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb97-39"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-39" tabindex="-1"></a>}</span>
<span id="cb97-40"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-40" tabindex="-1"></a>response <span class="ot">&lt;-</span> <span class="fu">renameNetwork</span>(<span class="at">title=</span>current_network_name, </span>
<span id="cb97-41"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-41" tabindex="-1"></a>                       <span class="at">network =</span> <span class="fu">as.numeric</span>(current_network_suid),</span>
<span id="cb97-42"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb97-42" tabindex="-1"></a>                       <span class="at">base.url =</span> current_base)</span></code></pre></div>
</div>
<div id="get-a-screen-shot-of-the-initial-network.-1" class="section level2 hasAnchor" number="9.9">
<h2><span class="header-section-number">9.9</span> Get a screen shot of the initial network.<a href="create-enrichment-map-from-r-with-gsea-results.html#get-a-screen-shot-of-the-initial-network.-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-1" tabindex="-1"></a><span class="co">#you can only output the file if it isn&#39;t on docker</span></span>
<span id="cb98-2"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-2" tabindex="-1"></a><span class="co">#on docker is put it into the user&#39;s home directory with docker </span></span>
<span id="cb98-3"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-3" tabindex="-1"></a><span class="co"># has not access to</span></span>
<span id="cb98-4"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-4" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span>is_docker){</span>
<span id="cb98-5"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-5" tabindex="-1"></a>  output_network_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">getwd</span>(),<span class="st">&quot;initial_screenshot_network.png&quot;</span>)</span>
<span id="cb98-6"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-6" tabindex="-1"></a>  output_network_file_current <span class="ot">&lt;-</span> output_network_file</span>
<span id="cb98-7"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-7" tabindex="-1"></a></span>
<span id="cb98-8"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-8" tabindex="-1"></a>  <span class="fu">fitContent</span>()</span>
<span id="cb98-9"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-9" tabindex="-1"></a></span>
<span id="cb98-10"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-10" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">file.exists</span>(output_network_file)){</span>
<span id="cb98-11"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-11" tabindex="-1"></a>    <span class="co">#cytoscape hangs waiting for user response if file already exists.</span></span>
<span id="cb98-12"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-12" tabindex="-1"></a>    <span class="co"># Remove it first</span></span>
<span id="cb98-13"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-13" tabindex="-1"></a>    response <span class="ot">&lt;-</span> <span class="fu">file.remove</span>(output_network_file)</span>
<span id="cb98-14"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-14" tabindex="-1"></a>  } </span>
<span id="cb98-15"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-15" tabindex="-1"></a></span>
<span id="cb98-16"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-16" tabindex="-1"></a>  response <span class="ot">&lt;-</span> <span class="fu">exportImage</span>(output_network_file, <span class="at">type =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb98-17"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-17" tabindex="-1"></a>                          <span class="at">base.url =</span> current_base)</span>
<span id="cb98-18"><a href="create-enrichment-map-from-r-with-gsea-results.html#cb98-18" tabindex="-1"></a>}</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="create-enrichment-map-from-r-with-gprofiler-results.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/risserlin/CBW_pathways_workshop_R_notebooks/edit/main/07-Create_EM_from_GSEA_results.Rmd",
"text": "Suggest an edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cbw_workshop_notebooks.pdf", "https://github.com/risserlin/CBW_pathways_workshop_R_notebooks/raw/main/07-Create_EM_from_GSEA_results.Rmd"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
