<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Run GSEA from within R | Translate gmt file from human to another species</title>
  <meta name="description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Run GSEA from within R | Translate gmt file from human to another species" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/cover.png" />
  <meta property="og:description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="github-repo" content="risserlin/CBW_pathways_workshop_R_notebooks" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Run GSEA from within R | Translate gmt file from human to another species" />
  
  <meta name="twitter:description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="twitter:image" content="/images/cover.png" />

<meta name="author" content="Ruth Isserlin" />


<meta name="date" content="2024-01-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="run-gprofiler-from-r.html"/>
<link rel="next" href="create-gmt-file-from-ensembl.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Workshop R Notebooks</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Index</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> CBW Workshop example R Notebooks</a></li>
<li class="chapter" data-level="3" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="setup.html"><a href="setup.html#install-r-and-rstudio"><i class="fa fa-check"></i><b>3.1</b> Install R and RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="setup.html"><a href="setup.html#docker-optional"><i class="fa fa-check"></i><b>3.2</b> Docker [Optional]</a></li>
<li class="chapter" data-level="3.3" data-path="setup.html"><a href="setup.html#install-docker"><i class="fa fa-check"></i><b>3.3</b> Install Docker</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="setup.html"><a href="setup.html#windows"><i class="fa fa-check"></i><b>3.3.1</b> Windows</a></li>
<li class="chapter" data-level="3.3.2" data-path="setup.html"><a href="setup.html#macos-linux"><i class="fa fa-check"></i><b>3.3.2</b> MacOS / Linux</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html"><i class="fa fa-check"></i><b>4</b> Run g:profiler from R</a>
<ul>
<li class="chapter" data-level="4.1" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#initialize-variables-and-libraries"><i class="fa fa-check"></i><b>4.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="4.2" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#load-in-query-set"><i class="fa fa-check"></i><b>4.2</b> Load in Query set</a></li>
<li class="chapter" data-level="4.3" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#run-gprofiler-with-supplied-genesets"><i class="fa fa-check"></i><b>4.3</b> Run g:profiler with supplied genesets</a></li>
<li class="chapter" data-level="4.4" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#download-and-load-gprofiler-geneset-file"><i class="fa fa-check"></i><b>4.4</b> Download and load g:profiler geneset file</a></li>
<li class="chapter" data-level="4.5" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#filter-results-by-geneset-size"><i class="fa fa-check"></i><b>4.5</b> Filter results by geneset size</a></li>
<li class="chapter" data-level="4.6" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#create-an-output-file-of-the-results---generic-enrichment-map-file-from-gprofiler-gmt"><i class="fa fa-check"></i><b>4.6</b> Create an output file of the results - Generic enrichment Map file from g:profiler gmt</a></li>
<li class="chapter" data-level="4.7" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#run-gprofiler-with-your-own-genesets-example-using-baderlab-genesets"><i class="fa fa-check"></i><b>4.7</b> Run g:profiler with your own genesets (example using BaderLab genesets)</a></li>
<li class="chapter" data-level="4.8" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#download-and-load-bader-lab-geneset-file"><i class="fa fa-check"></i><b>4.8</b> Download and load Bader lab geneset file</a></li>
<li class="chapter" data-level="4.9" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#filter-bader-lab-geneset-file"><i class="fa fa-check"></i><b>4.9</b> Filter Bader lab geneset file</a></li>
<li class="chapter" data-level="4.10" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#upload-the-gmt-files-to-gprofiler"><i class="fa fa-check"></i><b>4.10</b> Upload the gmt files to gprofiler</a></li>
<li class="chapter" data-level="4.11" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#create-an-output-file-of-the-results---generic-enrichment-map-file-from-baderlab-gmt"><i class="fa fa-check"></i><b>4.11</b> Create an output file of the results - Generic enrichment Map file from Baderlab gmt</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html"><i class="fa fa-check"></i><b>5</b> Run GSEA from within R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#load-in-required-libraries"><i class="fa fa-check"></i><b>5.1</b> Load in required libraries</a></li>
<li class="chapter" data-level="5.2" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#configurable-parameters"><i class="fa fa-check"></i><b>5.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#download-the-latest-pathway-definition-file"><i class="fa fa-check"></i><b>5.3</b> Download the latest pathway definition file</a></li>
<li class="chapter" data-level="5.4" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#run-gsea"><i class="fa fa-check"></i><b>5.4</b> Run GSEA</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html"><i class="fa fa-check"></i><b>6</b> Create GMT file from Ensembl</a>
<ul>
<li class="chapter" data-level="6.1" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#load-libraries"><i class="fa fa-check"></i><b>6.1</b> Load Libraries</a></li>
<li class="chapter" data-level="6.2" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#set-up-biomart-connection"><i class="fa fa-check"></i><b>6.2</b> Set up Biomart connection</a></li>
<li class="chapter" data-level="6.3" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#get-species-go-annotations"><i class="fa fa-check"></i><b>6.3</b> Get species GO annotations</a></li>
<li class="chapter" data-level="6.4" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#format-results-into-gmt-file"><i class="fa fa-check"></i><b>6.4</b> Format results into GMT file</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html"><i class="fa fa-check"></i><b>7</b> Translate human gmt file to another species using ensembl</a>
<ul>
<li class="chapter" data-level="7.1" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#load-in-required-libraries-1"><i class="fa fa-check"></i><b>7.1</b> Load in required libraries</a></li>
<li class="chapter" data-level="7.2" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#download-the-latest-pathway-definition-file-1"><i class="fa fa-check"></i><b>7.2</b> Download the latest pathway definition file</a></li>
<li class="chapter" data-level="7.3" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#set-up-biomart-connection-1"><i class="fa fa-check"></i><b>7.3</b> Set up Biomart connection</a></li>
<li class="chapter" data-level="7.4" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#convert-the-species-genes-from-human"><i class="fa fa-check"></i><b>7.4</b> Convert the species genes from human</a></li>
<li class="chapter" data-level="7.5" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#format-results-into-gmt-file-1"><i class="fa fa-check"></i><b>7.5</b> Format results into GMT file</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html"><i class="fa fa-check"></i><b>8</b> Create Enrichment map from R with g:Profiler results</a>
<ul>
<li class="chapter" data-level="8.1" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#initialize-variables-and-libraries-1"><i class="fa fa-check"></i><b>8.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="8.2" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#configurable-parameters-1"><i class="fa fa-check"></i><b>8.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="8.3" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#specify-data-files"><i class="fa fa-check"></i><b>8.3</b> Specify Data files</a></li>
<li class="chapter" data-level="8.4" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#launch-cytoscape"><i class="fa fa-check"></i><b>8.4</b> Launch Cytoscape</a></li>
<li class="chapter" data-level="8.5" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#make-sure-you-can-connect-to-cytoscape"><i class="fa fa-check"></i><b>8.5</b> Make sure you can connect to Cytoscape</a></li>
<li class="chapter" data-level="8.6" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#create-an-enrichment-map"><i class="fa fa-check"></i><b>8.6</b> Create an Enrichment map</a></li>
<li class="chapter" data-level="8.7" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#create-an-enrichment-map---run-em-command"><i class="fa fa-check"></i><b>8.7</b> Create an Enrichment map - run EM command</a></li>
<li class="chapter" data-level="8.8" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#get-a-screen-shot-of-the-initial-network."><i class="fa fa-check"></i><b>8.8</b> Get a screen shot of the initial network.</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html"><i class="fa fa-check"></i><b>9</b> Create Enrichment map from R with GSEA results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#initialize-variables-and-libraries-2"><i class="fa fa-check"></i><b>9.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="9.2" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#configurable-parameters-2"><i class="fa fa-check"></i><b>9.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="9.3" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#specify-data-files-1"><i class="fa fa-check"></i><b>9.3</b> Specify Data files</a></li>
<li class="chapter" data-level="9.4" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#optional-file-specification"><i class="fa fa-check"></i><b>9.4</b> Optional File specification</a></li>
<li class="chapter" data-level="9.5" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#launch-cytoscape-1"><i class="fa fa-check"></i><b>9.5</b> Launch Cytoscape</a></li>
<li class="chapter" data-level="9.6" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#make-sure-you-can-connect-to-cytoscape-1"><i class="fa fa-check"></i><b>9.6</b> Make sure you can connect to Cytoscape</a></li>
<li class="chapter" data-level="9.7" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map-1"><i class="fa fa-check"></i><b>9.7</b> Create an Enrichment map</a></li>
<li class="chapter" data-level="9.8" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map---run-em-command-1"><i class="fa fa-check"></i><b>9.8</b> Create an Enrichment map - run EM command</a></li>
<li class="chapter" data-level="9.9" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#get-a-screen-shot-of-the-initial-network.-1"><i class="fa fa-check"></i><b>9.9</b> Get a screen shot of the initial network.</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>10</b> References</a></li>
<li class="divider"></li>
<li><a href="https://baderlab.org" target="blank"><img src="./images/baderlablogo.png" width="280">Bader Lab</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Translate gmt file from human to another species</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="run-gsea-from-within-r" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Run GSEA from within R<a href="run-gsea-from-within-r.html#run-gsea-from-within-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This notebook is based largely on the <a href="https://baderlab.github.io/Cytoscape_workflows/EnrichmentMapPipeline/Protocol2_createEM.html">original notebook</a> published with EnrichmentMap Protocol<span class="citation">(<a href="#ref-em2019">Reimand et al. 2019</a>)</span></p>
<p>There is no package to run the original algorithm of GSEA<span class="citation">(<a href="#ref-gsea2005">Subramanian et al. 2005</a>)</span> in R. There are many packages that have been published to imitate the process but none are recognized by The GSEA team.<br />
<img src="images/gsea_r_package_message.png" alt="GSEA R package message" /></p>
<div id="load-in-required-libraries" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Load in required libraries<a href="run-gsea-from-within-r.html#load-in-required-libraries" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="run-gsea-from-within-r.html#cb43-1" tabindex="-1"></a><span class="co">#install required R and bioconductor packages</span></span>
<span id="cb43-2"><a href="run-gsea-from-within-r.html#cb43-2" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;RCurl&quot;</span>)}, </span>
<span id="cb43-3"><a href="run-gsea-from-within-r.html#cb43-3" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) {  </span>
<span id="cb43-4"><a href="run-gsea-from-within-r.html#cb43-4" tabindex="-1"></a>           <span class="fu">install.packages</span>(<span class="st">&quot;RCurl&quot;</span>)}, </span>
<span id="cb43-5"><a href="run-gsea-from-within-r.html#cb43-5" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;RCurl&quot;</span>))</span></code></pre></div>
</div>
<div id="configurable-parameters" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Configurable Parameters<a href="run-gsea-from-within-r.html#configurable-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to run GSEA automatically through the notebook you will need to download the gsea jar from <a href="http://software.broadinstitute.org/gsea/downloads.jsp">here</a>. Specify the exact path to the gsea jar in the parameters in order to automatically compute enrichments using GSEA.</p>
<p>If you are running this notebook using the <a href="https://hub.docker.com/r/risserlin/workshop_base_image">baderlab workshop docker image</a> then the image comes pre-installed with the gsea jar that you can use to run gsea directly in the docker. The path to the GSEA jar in the docker is - /home/rstudio/GSEA_4.3.2/gsea-cli.sh</p>
<p>In order to run GSEA automatically you need to speciry the path to the gsea jar file.
The gsea_jar needs to be the full path to the GSEA 4.3.2 directory that you downloaded from GSEA. for example /Users/johnsmith/GSEA_4.3.2/gsea-cli.sh</p>
<p>The parameters are set manually here but if you want to run the script from the command line then you can update the notebook to pull the parameters from the command line given arguments by updating each variable below to pull the values from the paramters - for example:</p>
<ul>
<li>variable &lt;- params$parameter_name</li>
</ul>
<p>For more details see - <a href="https://bookdown.org/yihui/rmarkdown/params-declare.html">defining and using parameters</a> and <a href="https://bookdown.org/yihui/rmarkdown/params-knit.html">Knitting with parameters</a></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="run-gsea-from-within-r.html#cb44-1" tabindex="-1"></a><span class="co">#path to GSEA jar </span></span>
<span id="cb44-2"><a href="run-gsea-from-within-r.html#cb44-2" tabindex="-1"></a>gsea_jar <span class="ot">&lt;-</span> <span class="st">&quot;/home/rstudio/GSEA_4.3.2/gsea-cli.sh&quot;</span></span></code></pre></div>
<p>Set the working directory as the directory to the directory where you downloaded all protocol files. For example /User/JohnSmith/EMProtocolFiles/data</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="run-gsea-from-within-r.html#cb45-1" tabindex="-1"></a><span class="co">#directory where all the data files are found.  For example -   ./data/ </span></span>
<span id="cb45-2"><a href="run-gsea-from-within-r.html#cb45-2" tabindex="-1"></a>working_dir <span class="ot">&lt;-</span> <span class="st">&quot;./data/&quot;</span></span>
<span id="cb45-3"><a href="run-gsea-from-within-r.html#cb45-3" tabindex="-1"></a></span>
<span id="cb45-4"><a href="run-gsea-from-within-r.html#cb45-4" tabindex="-1"></a><span class="co">#directory where all the data files are found.  For example -   ./generated_data/gsea/</span></span>
<span id="cb45-5"><a href="run-gsea-from-within-r.html#cb45-5" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">&quot;./generated_data/gsea/&quot;</span></span>
<span id="cb45-6"><a href="run-gsea-from-within-r.html#cb45-6" tabindex="-1"></a></span>
<span id="cb45-7"><a href="run-gsea-from-within-r.html#cb45-7" tabindex="-1"></a><span class="co">#The name to give the analysis in GSEA - for example Mesen_vs_Immuno</span></span>
<span id="cb45-8"><a href="run-gsea-from-within-r.html#cb45-8" tabindex="-1"></a>analysis_name <span class="ot">&lt;-</span> <span class="st">&quot;Mesen_vs_Immuno&quot;</span></span>
<span id="cb45-9"><a href="run-gsea-from-within-r.html#cb45-9" tabindex="-1"></a></span>
<span id="cb45-10"><a href="run-gsea-from-within-r.html#cb45-10" tabindex="-1"></a><span class="co">#rank file to use in GSEA analysis.  </span></span>
<span id="cb45-11"><a href="run-gsea-from-within-r.html#cb45-11" tabindex="-1"></a><span class="co">#For example - MesenchymalvsImmunoreactive_edger_ranks.rnk</span></span>
<span id="cb45-12"><a href="run-gsea-from-within-r.html#cb45-12" tabindex="-1"></a>rnk_file <span class="ot">&lt;-</span> <span class="st">&quot;MesenchymalvsImmunoreactive_edger_ranks.rnk&quot;</span></span>
<span id="cb45-13"><a href="run-gsea-from-within-r.html#cb45-13" tabindex="-1"></a></span>
<span id="cb45-14"><a href="run-gsea-from-within-r.html#cb45-14" tabindex="-1"></a><span class="co">#run_gsea - true/false</span></span>
<span id="cb45-15"><a href="run-gsea-from-within-r.html#cb45-15" tabindex="-1"></a><span class="co"># This parameter is for the compilation of the notebook.  </span></span>
<span id="cb45-16"><a href="run-gsea-from-within-r.html#cb45-16" tabindex="-1"></a>run_gsea <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb45-17"><a href="run-gsea-from-within-r.html#cb45-17" tabindex="-1"></a></span>
<span id="cb45-18"><a href="run-gsea-from-within-r.html#cb45-18" tabindex="-1"></a><span class="co">#set the gmt file you want to use if you don&#39;t want to use the latest gmt file.</span></span>
<span id="cb45-19"><a href="run-gsea-from-within-r.html#cb45-19" tabindex="-1"></a><span class="co"># For example, if you set dest_gmt_file ==&quot;&quot; the below script will automatically</span></span>
<span id="cb45-20"><a href="run-gsea-from-within-r.html#cb45-20" tabindex="-1"></a><span class="co"># download the latest gmt file from baderlab webstie.  If it is set then it</span></span>
<span id="cb45-21"><a href="run-gsea-from-within-r.html#cb45-21" tabindex="-1"></a><span class="co"># will use the file specified.  </span></span>
<span id="cb45-22"><a href="run-gsea-from-within-r.html#cb45-22" tabindex="-1"></a>dest_gmt_file <span class="ot">=</span> <span class="fu">file.path</span>(output_dir, </span>
<span id="cb45-23"><a href="run-gsea-from-within-r.html#cb45-23" tabindex="-1"></a>                          <span class="st">&quot;Human_GOBP_AllPathways_no_GO_iea_April_02_2023_symbol.gmt&quot;</span>)</span></code></pre></div>
</div>
<div id="download-the-latest-pathway-definition-file" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Download the latest pathway definition file<a href="run-gsea-from-within-r.html#download-the-latest-pathway-definition-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Only Human, Mouse, Rat, and Woodchuck gene set files are currently available on the baderlab downloads site. If you are working with a species other than human (and it is either rat,mouse or woodchuck) change the gmt_url below to the correct species. Check <a href="http://download.baderlab.org/EM_Genesets/current_release/">here</a> to see all available species.</p>
<p>To create your own GMT file using Ensembl see <a href="create-gmt-file-from-ensembl.html#create-gmt-file-from-ensembl">Create GMT file from Ensembl</a></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="run-gsea-from-within-r.html#cb46-1" tabindex="-1"></a><span class="cf">if</span>(dest_gmt_file <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb46-2"><a href="run-gsea-from-within-r.html#cb46-2" tabindex="-1"></a>  gmt_url <span class="ot">=</span> <span class="st">&quot;http://download.baderlab.org/EM_Genesets/current_release/Human/symbol/&quot;</span></span>
<span id="cb46-3"><a href="run-gsea-from-within-r.html#cb46-3" tabindex="-1"></a>  </span>
<span id="cb46-4"><a href="run-gsea-from-within-r.html#cb46-4" tabindex="-1"></a>  <span class="co">#list all the files on the server</span></span>
<span id="cb46-5"><a href="run-gsea-from-within-r.html#cb46-5" tabindex="-1"></a>  filenames <span class="ot">=</span> <span class="fu">getURL</span>(gmt_url)</span>
<span id="cb46-6"><a href="run-gsea-from-within-r.html#cb46-6" tabindex="-1"></a>  tc <span class="ot">=</span> <span class="fu">textConnection</span>(filenames)</span>
<span id="cb46-7"><a href="run-gsea-from-within-r.html#cb46-7" tabindex="-1"></a>  contents <span class="ot">=</span> <span class="fu">readLines</span>(tc)</span>
<span id="cb46-8"><a href="run-gsea-from-within-r.html#cb46-8" tabindex="-1"></a>  <span class="fu">close</span>(tc)</span>
<span id="cb46-9"><a href="run-gsea-from-within-r.html#cb46-9" tabindex="-1"></a>  </span>
<span id="cb46-10"><a href="run-gsea-from-within-r.html#cb46-10" tabindex="-1"></a>  <span class="co">#get the gmt that has all the pathways and does not include terms </span></span>
<span id="cb46-11"><a href="run-gsea-from-within-r.html#cb46-11" tabindex="-1"></a>  <span class="co"># inferred from electronic annotations(IEA)</span></span>
<span id="cb46-12"><a href="run-gsea-from-within-r.html#cb46-12" tabindex="-1"></a>  <span class="co">#start with gmt file that has pathways only and GO Biological Process only.</span></span>
<span id="cb46-13"><a href="run-gsea-from-within-r.html#cb46-13" tabindex="-1"></a>  rx <span class="ot">=</span> <span class="fu">gregexpr</span>(<span class="st">&quot;(?&lt;=&lt;a href=</span><span class="sc">\&quot;</span><span class="st">)(.*.GOBP_AllPathways_no_GO_iea.*.)(.gmt)(?=</span><span class="sc">\&quot;</span><span class="st">&gt;)&quot;</span>,</span>
<span id="cb46-14"><a href="run-gsea-from-within-r.html#cb46-14" tabindex="-1"></a>    contents, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb46-15"><a href="run-gsea-from-within-r.html#cb46-15" tabindex="-1"></a>  gmt_file <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">regmatches</span>(contents, rx))</span>
<span id="cb46-16"><a href="run-gsea-from-within-r.html#cb46-16" tabindex="-1"></a>  </span>
<span id="cb46-17"><a href="run-gsea-from-within-r.html#cb46-17" tabindex="-1"></a>  dest_gmt_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir,gmt_file )</span>
<span id="cb46-18"><a href="run-gsea-from-within-r.html#cb46-18" tabindex="-1"></a>  </span>
<span id="cb46-19"><a href="run-gsea-from-within-r.html#cb46-19" tabindex="-1"></a>  <span class="co">#check if this gmt file already exists</span></span>
<span id="cb46-20"><a href="run-gsea-from-within-r.html#cb46-20" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(dest_gmt_file)){</span>
<span id="cb46-21"><a href="run-gsea-from-within-r.html#cb46-21" tabindex="-1"></a>    <span class="fu">download.file</span>(</span>
<span id="cb46-22"><a href="run-gsea-from-within-r.html#cb46-22" tabindex="-1"></a>      <span class="fu">paste</span>(gmt_url,gmt_file,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb46-23"><a href="run-gsea-from-within-r.html#cb46-23" tabindex="-1"></a>      <span class="at">destfile=</span>dest_gmt_file</span>
<span id="cb46-24"><a href="run-gsea-from-within-r.html#cb46-24" tabindex="-1"></a>    )</span>
<span id="cb46-25"><a href="run-gsea-from-within-r.html#cb46-25" tabindex="-1"></a>  }</span>
<span id="cb46-26"><a href="run-gsea-from-within-r.html#cb46-26" tabindex="-1"></a>}</span></code></pre></div>
<hr />
</div>
<div id="run-gsea" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Run GSEA<a href="run-gsea-from-within-r.html#run-gsea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>(GSEA)[<a href="http://software.broadinstitute.org/gsea/index.jsp" class="uri">http://software.broadinstitute.org/gsea/index.jsp</a>] is a stand alone java program with many customizable options. It can be easily run through its integrated user interface. To make this a seemless pipeline we can run GSEA from the command line with a set of options. Any of the supplied options can be customized and there are many additional options that can be specified. For more details see (here)[<a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Running_GSEA_from" class="uri">http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Running_GSEA_from</a>]</p>
<p>In the below command the following options have been specified:</p>
<ul>
<li>rnk - path to the rank file</li>
<li>gmx - path to the gene set definition (gmt) file</li>
<li>collapse - true/false indicates whether the expression/rnk file needs to be collapsed from probes to gene symbols</li>
<li>nperm - number of permutations</li>
<li>scoring_scheme -</li>
<li>rpt_label - name of the directory with output</li>
<li>rnd_seed - random seed to use</li>
<li>set_max - maximum size for individual gene sets. In GSEA interface this is set to 500 but we prefer to use a more stringent setting of 200.</li>
<li>set_min - minimum size for individual gene sets</li>
<li>zip_report - true/false to zip output directory</li>
<li>out - directory where to place the result directory.</li>
</ul>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="run-gsea-from-within-r.html#cb47-1" tabindex="-1"></a><span class="cf">if</span>(run_gsea){</span>
<span id="cb47-2"><a href="run-gsea-from-within-r.html#cb47-2" tabindex="-1"></a>  command <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;&quot;</span>,gsea_jar,  </span>
<span id="cb47-3"><a href="run-gsea-from-within-r.html#cb47-3" tabindex="-1"></a>                   <span class="st">&quot;GSEAPreRanked -gmx&quot;</span>, dest_gmt_file, </span>
<span id="cb47-4"><a href="run-gsea-from-within-r.html#cb47-4" tabindex="-1"></a>                   <span class="st">&quot;-rnk&quot;</span> ,<span class="fu">file.path</span>(working_dir,rnk_file), </span>
<span id="cb47-5"><a href="run-gsea-from-within-r.html#cb47-5" tabindex="-1"></a>                   <span class="st">&quot;-collapse false -nperm 1000 -scoring_scheme weighted&quot;</span>, </span>
<span id="cb47-6"><a href="run-gsea-from-within-r.html#cb47-6" tabindex="-1"></a>                   <span class="st">&quot;-rpt_label &quot;</span>,analysis_name,</span>
<span id="cb47-7"><a href="run-gsea-from-within-r.html#cb47-7" tabindex="-1"></a>                   <span class="st">&quot;  -plot_top_x 20 -rnd_seed 12345  -set_max 200&quot;</span>,  </span>
<span id="cb47-8"><a href="run-gsea-from-within-r.html#cb47-8" tabindex="-1"></a>                   <span class="st">&quot; -set_min 15 -zip_report false &quot;</span>,</span>
<span id="cb47-9"><a href="run-gsea-from-within-r.html#cb47-9" tabindex="-1"></a>                   <span class="st">&quot; -out&quot;</span> ,output_dir, </span>
<span id="cb47-10"><a href="run-gsea-from-within-r.html#cb47-10" tabindex="-1"></a>                   <span class="st">&quot; &gt; gsea_output.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb47-11"><a href="run-gsea-from-within-r.html#cb47-11" tabindex="-1"></a>  <span class="fu">system</span>(command)</span>
<span id="cb47-12"><a href="run-gsea-from-within-r.html#cb47-12" tabindex="-1"></a>}</span></code></pre></div>

</div>
</div>
<h3> References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-em2019" class="csl-entry">
Reimand, Jüri, Ruth Isserlin, Veronique Voisin, Mike Kucera, Christian Tannus-Lopes, Asha Rostamianfar, Lina Wadi, et al. 2019. <span>“Pathway Enrichment Analysis and Visualization of Omics Data Using g: Profiler, GSEA, Cytoscape and EnrichmentMap.”</span> <em>Nature Protocols</em> 14 (2): 482–517.
</div>
<div id="ref-gsea2005" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, et al. 2005. <span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="run-gprofiler-from-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="create-gmt-file-from-ensembl.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/risserlin/CBW_pathways_workshop_R_notebooks/edit/main/04-Run_gsea_in_R.Rmd",
"text": "Suggest an edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cbw_workshop_notebooks.pdf", "https://github.com/risserlin/CBW_pathways_workshop_R_notebooks/raw/main/04-Run_gsea_in_R.Rmd"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
