<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Translate human gmt file to another species using ensembl | Translate gmt file from human to another species</title>
  <meta name="description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Translate human gmt file to another species using ensembl | Translate gmt file from human to another species" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/images/cover.png" />
  <meta property="og:description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="github-repo" content="risserlin/CBW_pathways_workshop_R_notebooks" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Translate human gmt file to another species using ensembl | Translate gmt file from human to another species" />
  
  <meta name="twitter:description" content="These are example R Notebooks to run g:profiler and GSEA and visualize results in Cytoscape with Enrichment map directly from R" />
  <meta name="twitter:image" content="/images/cover.png" />

<meta name="author" content="Ruth Isserlin" />


<meta name="date" content="2024-01-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="create-gmt-file-from-ensembl.html"/>
<link rel="next" href="create-enrichment-map-from-r-with-gprofiler-results.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Workshop R Notebooks</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Index</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> CBW Workshop example R Notebooks</a></li>
<li class="chapter" data-level="3" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="setup.html"><a href="setup.html#install-r-and-rstudio"><i class="fa fa-check"></i><b>3.1</b> Install R and RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="setup.html"><a href="setup.html#docker-optional"><i class="fa fa-check"></i><b>3.2</b> Docker [Optional]</a></li>
<li class="chapter" data-level="3.3" data-path="setup.html"><a href="setup.html#install-docker"><i class="fa fa-check"></i><b>3.3</b> Install Docker</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="setup.html"><a href="setup.html#windows"><i class="fa fa-check"></i><b>3.3.1</b> Windows</a></li>
<li class="chapter" data-level="3.3.2" data-path="setup.html"><a href="setup.html#macos-linux"><i class="fa fa-check"></i><b>3.3.2</b> MacOS / Linux</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html"><i class="fa fa-check"></i><b>4</b> Run g:profiler from R</a>
<ul>
<li class="chapter" data-level="4.1" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#initialize-variables-and-libraries"><i class="fa fa-check"></i><b>4.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="4.2" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#load-in-query-set"><i class="fa fa-check"></i><b>4.2</b> Load in Query set</a></li>
<li class="chapter" data-level="4.3" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#run-gprofiler-with-supplied-genesets"><i class="fa fa-check"></i><b>4.3</b> Run g:profiler with supplied genesets</a></li>
<li class="chapter" data-level="4.4" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#download-and-load-gprofiler-geneset-file"><i class="fa fa-check"></i><b>4.4</b> Download and load g:profiler geneset file</a></li>
<li class="chapter" data-level="4.5" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#filter-results-by-geneset-size"><i class="fa fa-check"></i><b>4.5</b> Filter results by geneset size</a></li>
<li class="chapter" data-level="4.6" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#create-an-output-file-of-the-results---generic-enrichment-map-file-from-gprofiler-gmt"><i class="fa fa-check"></i><b>4.6</b> Create an output file of the results - Generic enrichment Map file from g:profiler gmt</a></li>
<li class="chapter" data-level="4.7" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#run-gprofiler-with-your-own-genesets-example-using-baderlab-genesets"><i class="fa fa-check"></i><b>4.7</b> Run g:profiler with your own genesets (example using BaderLab genesets)</a></li>
<li class="chapter" data-level="4.8" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#download-and-load-bader-lab-geneset-file"><i class="fa fa-check"></i><b>4.8</b> Download and load Bader lab geneset file</a></li>
<li class="chapter" data-level="4.9" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#filter-bader-lab-geneset-file"><i class="fa fa-check"></i><b>4.9</b> Filter Bader lab geneset file</a></li>
<li class="chapter" data-level="4.10" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#upload-the-gmt-files-to-gprofiler"><i class="fa fa-check"></i><b>4.10</b> Upload the gmt files to gprofiler</a></li>
<li class="chapter" data-level="4.11" data-path="run-gprofiler-from-r.html"><a href="run-gprofiler-from-r.html#create-an-output-file-of-the-results---generic-enrichment-map-file-from-baderlab-gmt"><i class="fa fa-check"></i><b>4.11</b> Create an output file of the results - Generic enrichment Map file from Baderlab gmt</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html"><i class="fa fa-check"></i><b>5</b> Run GSEA from within R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#load-in-required-libraries"><i class="fa fa-check"></i><b>5.1</b> Load in required libraries</a></li>
<li class="chapter" data-level="5.2" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#configurable-parameters"><i class="fa fa-check"></i><b>5.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="5.3" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#download-the-latest-pathway-definition-file"><i class="fa fa-check"></i><b>5.3</b> Download the latest pathway definition file</a></li>
<li class="chapter" data-level="5.4" data-path="run-gsea-from-within-r.html"><a href="run-gsea-from-within-r.html#run-gsea"><i class="fa fa-check"></i><b>5.4</b> Run GSEA</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html"><i class="fa fa-check"></i><b>6</b> Create GMT file from Ensembl</a>
<ul>
<li class="chapter" data-level="6.1" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#load-libraries"><i class="fa fa-check"></i><b>6.1</b> Load Libraries</a></li>
<li class="chapter" data-level="6.2" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#set-up-biomart-connection"><i class="fa fa-check"></i><b>6.2</b> Set up Biomart connection</a></li>
<li class="chapter" data-level="6.3" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#get-species-go-annotations"><i class="fa fa-check"></i><b>6.3</b> Get species GO annotations</a></li>
<li class="chapter" data-level="6.4" data-path="create-gmt-file-from-ensembl.html"><a href="create-gmt-file-from-ensembl.html#format-results-into-gmt-file"><i class="fa fa-check"></i><b>6.4</b> Format results into GMT file</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html"><i class="fa fa-check"></i><b>7</b> Translate human gmt file to another species using ensembl</a>
<ul>
<li class="chapter" data-level="7.1" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#load-in-required-libraries-1"><i class="fa fa-check"></i><b>7.1</b> Load in required libraries</a></li>
<li class="chapter" data-level="7.2" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#download-the-latest-pathway-definition-file-1"><i class="fa fa-check"></i><b>7.2</b> Download the latest pathway definition file</a></li>
<li class="chapter" data-level="7.3" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#set-up-biomart-connection-1"><i class="fa fa-check"></i><b>7.3</b> Set up Biomart connection</a></li>
<li class="chapter" data-level="7.4" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#convert-the-species-genes-from-human"><i class="fa fa-check"></i><b>7.4</b> Convert the species genes from human</a></li>
<li class="chapter" data-level="7.5" data-path="translate-human-gmt-file-to-another-species-using-ensembl.html"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#format-results-into-gmt-file-1"><i class="fa fa-check"></i><b>7.5</b> Format results into GMT file</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html"><i class="fa fa-check"></i><b>8</b> Create Enrichment map from R with g:Profiler results</a>
<ul>
<li class="chapter" data-level="8.1" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#initialize-variables-and-libraries-1"><i class="fa fa-check"></i><b>8.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="8.2" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#configurable-parameters-1"><i class="fa fa-check"></i><b>8.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="8.3" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#specify-data-files"><i class="fa fa-check"></i><b>8.3</b> Specify Data files</a></li>
<li class="chapter" data-level="8.4" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#launch-cytoscape"><i class="fa fa-check"></i><b>8.4</b> Launch Cytoscape</a></li>
<li class="chapter" data-level="8.5" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#make-sure-you-can-connect-to-cytoscape"><i class="fa fa-check"></i><b>8.5</b> Make sure you can connect to Cytoscape</a></li>
<li class="chapter" data-level="8.6" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#create-an-enrichment-map"><i class="fa fa-check"></i><b>8.6</b> Create an Enrichment map</a></li>
<li class="chapter" data-level="8.7" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#create-an-enrichment-map---run-em-command"><i class="fa fa-check"></i><b>8.7</b> Create an Enrichment map - run EM command</a></li>
<li class="chapter" data-level="8.8" data-path="create-enrichment-map-from-r-with-gprofiler-results.html"><a href="create-enrichment-map-from-r-with-gprofiler-results.html#get-a-screen-shot-of-the-initial-network."><i class="fa fa-check"></i><b>8.8</b> Get a screen shot of the initial network.</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html"><i class="fa fa-check"></i><b>9</b> Create Enrichment map from R with GSEA results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#initialize-variables-and-libraries-2"><i class="fa fa-check"></i><b>9.1</b> Initialize variables and libraries</a></li>
<li class="chapter" data-level="9.2" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#configurable-parameters-2"><i class="fa fa-check"></i><b>9.2</b> Configurable Parameters</a></li>
<li class="chapter" data-level="9.3" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#specify-data-files-1"><i class="fa fa-check"></i><b>9.3</b> Specify Data files</a></li>
<li class="chapter" data-level="9.4" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#optional-file-specification"><i class="fa fa-check"></i><b>9.4</b> Optional File specification</a></li>
<li class="chapter" data-level="9.5" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#launch-cytoscape-1"><i class="fa fa-check"></i><b>9.5</b> Launch Cytoscape</a></li>
<li class="chapter" data-level="9.6" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#make-sure-you-can-connect-to-cytoscape-1"><i class="fa fa-check"></i><b>9.6</b> Make sure you can connect to Cytoscape</a></li>
<li class="chapter" data-level="9.7" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map-1"><i class="fa fa-check"></i><b>9.7</b> Create an Enrichment map</a></li>
<li class="chapter" data-level="9.8" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#create-an-enrichment-map---run-em-command-1"><i class="fa fa-check"></i><b>9.8</b> Create an Enrichment map - run EM command</a></li>
<li class="chapter" data-level="9.9" data-path="create-enrichment-map-from-r-with-gsea-results.html"><a href="create-enrichment-map-from-r-with-gsea-results.html#get-a-screen-shot-of-the-initial-network.-1"><i class="fa fa-check"></i><b>9.9</b> Get a screen shot of the initial network.</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>10</b> References</a></li>
<li class="divider"></li>
<li><a href="https://baderlab.org" target="blank"><img src="./images/baderlablogo.png" width="280">Bader Lab</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Translate gmt file from human to another species</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="translate-human-gmt-file-to-another-species-using-ensembl" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Translate human gmt file to another species using ensembl<a href="translate-human-gmt-file-to-another-species-using-ensembl.html#translate-human-gmt-file-to-another-species-using-ensembl" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>for species where there is a not an existing effort to characterize genes through GO, a much richer gmt file will come from mapping the gmt file from a well defined species preferrably the species your experiment is trying to model (for example, if you are modelling a human disease in mouse using a human gmt might return much richer results but species specific changes might be lost )</p>
<div id="load-in-required-libraries-1" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Load in required libraries<a href="translate-human-gmt-file-to-another-species-using-ensembl.html#load-in-required-libraries-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-1" tabindex="-1"></a><span class="co">#install required R and bioconductor packages</span></span>
<span id="cb64-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-2" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;RCurl&quot;</span>)}, </span>
<span id="cb64-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-3" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) {  </span>
<span id="cb64-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-4" tabindex="-1"></a>           <span class="fu">install.packages</span>(<span class="st">&quot;RCurl&quot;</span>)}, </span>
<span id="cb64-5"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-5" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;RCurl&quot;</span>))</span>
<span id="cb64-6"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-6" tabindex="-1"></a></span>
<span id="cb64-7"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-7" tabindex="-1"></a><span class="co">#use library</span></span>
<span id="cb64-8"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-8" tabindex="-1"></a><span class="co">#make sure biocManager is installed</span></span>
<span id="cb64-9"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-9" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;BiocManager&quot;</span>)}, </span>
<span id="cb64-10"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-10" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) { </span>
<span id="cb64-11"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-11" tabindex="-1"></a>           <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)}, </span>
<span id="cb64-12"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-12" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;BiocManager&quot;</span>))</span>
<span id="cb64-13"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-13" tabindex="-1"></a></span>
<span id="cb64-14"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-14" tabindex="-1"></a></span>
<span id="cb64-15"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-15" tabindex="-1"></a><span class="fu">tryCatch</span>(<span class="at">expr =</span> { <span class="fu">library</span>(<span class="st">&quot;biomaRt&quot;</span>)}, </span>
<span id="cb64-16"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-16" tabindex="-1"></a>         <span class="at">error =</span> <span class="cf">function</span>(e) { </span>
<span id="cb64-17"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-17" tabindex="-1"></a>           BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;biomaRt&quot;</span>)}, </span>
<span id="cb64-18"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb64-18" tabindex="-1"></a>         <span class="at">finally =</span> <span class="fu">library</span>(<span class="st">&quot;biomaRt&quot;</span>))</span></code></pre></div>
</div>
<div id="download-the-latest-pathway-definition-file-1" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Download the latest pathway definition file<a href="translate-human-gmt-file-to-another-species-using-ensembl.html#download-the-latest-pathway-definition-file-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Only Human, Mouse, Rat, and Woodchuck gene set files are currently available on the baderlab downloads site. If you are working with a species other than human (and it is either rat,mouse or woodchuck) change the gmt_url below to the correct species. Check <a href="http://download.baderlab.org/EM_Genesets/current_release/">here</a> to see all available species.</p>
<p>To create your own GMT file using Ensembl see <a href="create-gmt-file-from-ensembl.html#create-gmt-file-from-ensembl">Create GMT file from Ensembl</a></p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-1" tabindex="-1"></a>dest_gmt_file <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span> <span class="co">#params$dest_gmt_file </span></span>
<span id="cb65-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-2" tabindex="-1"></a></span>
<span id="cb65-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-3" tabindex="-1"></a><span class="cf">if</span>(dest_gmt_file <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb65-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-4" tabindex="-1"></a>  gmt_url <span class="ot">=</span> <span class="st">&quot;http://download.baderlab.org/EM_Genesets/current_release/Human/symbol/&quot;</span></span>
<span id="cb65-5"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-5" tabindex="-1"></a>  </span>
<span id="cb65-6"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-6" tabindex="-1"></a>  <span class="co">#list all the files on the server</span></span>
<span id="cb65-7"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-7" tabindex="-1"></a>  filenames <span class="ot">=</span> <span class="fu">getURL</span>(gmt_url)</span>
<span id="cb65-8"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-8" tabindex="-1"></a>  tc <span class="ot">=</span> <span class="fu">textConnection</span>(filenames)</span>
<span id="cb65-9"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-9" tabindex="-1"></a>  contents <span class="ot">=</span> <span class="fu">readLines</span>(tc)</span>
<span id="cb65-10"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-10" tabindex="-1"></a>  <span class="fu">close</span>(tc)</span>
<span id="cb65-11"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-11" tabindex="-1"></a>  </span>
<span id="cb65-12"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-12" tabindex="-1"></a>  <span class="co">#get the gmt that has all the pathways and does not include terms </span></span>
<span id="cb65-13"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-13" tabindex="-1"></a>  <span class="co"># inferred from electronic annotations(IEA)</span></span>
<span id="cb65-14"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-14" tabindex="-1"></a>  <span class="co">#start with gmt file that has pathways only and GO Biological Process only.</span></span>
<span id="cb65-15"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-15" tabindex="-1"></a>  rx <span class="ot">=</span> <span class="fu">gregexpr</span>(<span class="st">&quot;(?&lt;=&lt;a href=</span><span class="sc">\&quot;</span><span class="st">)(.*.GOBP_AllPathways_no_GO_iea.*.)(.gmt)(?=</span><span class="sc">\&quot;</span><span class="st">&gt;)&quot;</span>,</span>
<span id="cb65-16"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-16" tabindex="-1"></a>    contents, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb65-17"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-17" tabindex="-1"></a>  gmt_file <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">regmatches</span>(contents, rx))</span>
<span id="cb65-18"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-18" tabindex="-1"></a>  </span>
<span id="cb65-19"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-19" tabindex="-1"></a>  dest_gmt_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(params<span class="sc">$</span>working_dir,gmt_file )</span>
<span id="cb65-20"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-20" tabindex="-1"></a>  </span>
<span id="cb65-21"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-21" tabindex="-1"></a>  <span class="co">#check if this gmt file already exists</span></span>
<span id="cb65-22"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-22" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(dest_gmt_file)){</span>
<span id="cb65-23"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-23" tabindex="-1"></a>    <span class="fu">download.file</span>(</span>
<span id="cb65-24"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-24" tabindex="-1"></a>      <span class="fu">paste</span>(gmt_url,gmt_file,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb65-25"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-25" tabindex="-1"></a>      <span class="at">destfile=</span>dest_gmt_file</span>
<span id="cb65-26"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-26" tabindex="-1"></a>    )</span>
<span id="cb65-27"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-27" tabindex="-1"></a>  }</span>
<span id="cb65-28"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb65-28" tabindex="-1"></a>}</span></code></pre></div>
<p>Load in the originaing gmt file</p>
</div>
<div id="set-up-biomart-connection-1" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Set up Biomart connection<a href="translate-human-gmt-file-to-another-species-using-ensembl.html#set-up-biomart-connection-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Connect to Biomart</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb66-1" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">&quot;genes&quot;</span> , <span class="at">host =</span> <span class="st">&quot;asia.ensembl.org&quot;</span>)</span>
<span id="cb66-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb66-2" tabindex="-1"></a><span class="co">#ensembl &lt;- useEnsembl(&quot;ensembl&quot;)</span></span></code></pre></div>
<p>Figure out which dataset you want to use - for some species there might be a few datasets to choose from. Not all of the datasets have common namesa associated with them. For example, if you search for ‘yeast’ nothing will be returned but if you look for Saccharomyces or cerevisiae you will be able to find it.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb67-1" tabindex="-1"></a>all_datasets <span class="ot">&lt;-</span> <span class="fu">listDatasets</span>(ensembl, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Attempting web service request:
## asia.ensembl.org:80/biomart/martservice?redirect=no&amp;type=datasets&amp;requestid=biomaRt&amp;mart=ENSEMBL_MART_ENSEMBL</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb69-1" tabindex="-1"></a><span class="co">#get all the datasets that match our species definition</span></span>
<span id="cb69-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb69-2" tabindex="-1"></a>all_datasets[<span class="fu">grep</span>(all_datasets<span class="sc">$</span>description,</span>
<span id="cb69-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb69-3" tabindex="-1"></a>                  <span class="at">pattern=</span><span class="fu">paste</span>(params<span class="sc">$</span>new_species,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb69-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb69-4" tabindex="-1"></a>                  <span class="at">ignore.case =</span> <span class="cn">TRUE</span>),]</span></code></pre></div>
<pre><code>##                dataset                 description         version
## 68 fcatus_gene_ensembl Cat genes (Felis_catus_9.0) Felis_catus_9.0</code></pre>
<p>Based on the above table define the dataset</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb71-1" tabindex="-1"></a><span class="co">#get all the datasets that match our species definition</span></span>
<span id="cb71-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb71-2" tabindex="-1"></a>base_dataset <span class="ot">&lt;-</span> all_datasets<span class="sc">$</span>dataset[<span class="fu">grep</span>(all_datasets<span class="sc">$</span>description,</span>
<span id="cb71-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb71-3" tabindex="-1"></a>                  <span class="at">pattern=</span><span class="fu">paste</span>(params<span class="sc">$</span>new_species,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb71-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb71-4" tabindex="-1"></a>                  <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)]</span></code></pre></div>
<p>If you know the ensembl dataset that you want to use you can specify it in the parameters above or grab from the above table the dataset of the species that you are interested in.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb72-1" tabindex="-1"></a>ensembl <span class="ot">=</span> <span class="fu">useDataset</span>(base_dataset,<span class="at">mart=</span>ensembl)</span></code></pre></div>
</div>
<div id="convert-the-species-genes-from-human" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Convert the species genes from human<a href="translate-human-gmt-file-to-another-species-using-ensembl.html#convert-the-species-genes-from-human" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Get the homologs in the species of interest of the human genes.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-1" tabindex="-1"></a>homologs <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;external_gene_name&quot;</span>,</span>
<span id="cb73-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-2" tabindex="-1"></a>                                      <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb73-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-3" tabindex="-1"></a>                                      <span class="st">&quot;hsapiens_homolog_ensembl_gene&quot;</span>, </span>
<span id="cb73-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-4" tabindex="-1"></a>                                      <span class="st">&quot;hsapiens_homolog_associated_gene_name&quot;</span> </span>
<span id="cb73-5"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-5" tabindex="-1"></a>                                      ), </span>
<span id="cb73-6"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-6" tabindex="-1"></a>                       <span class="at">filters=</span><span class="fu">list</span>(<span class="at">biotype=</span><span class="st">&#39;protein_coding&#39;</span>), <span class="at">mart=</span>ensembl);</span>
<span id="cb73-7"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-7" tabindex="-1"></a><span class="co"># get rid of rows that have an empty gene name for new species or for human</span></span>
<span id="cb73-8"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb73-8" tabindex="-1"></a>homologs <span class="ot">&lt;-</span> homologs[<span class="fu">which</span>((homologs<span class="sc">$</span>external_gene_name <span class="sc">!=</span> <span class="st">&quot;&quot;</span>) <span class="sc">&amp;</span> (homologs<span class="sc">$</span>hsapiens_homolog_associated_gene_name <span class="sc">!=</span> <span class="st">&quot;&quot;</span>)),]</span></code></pre></div>
<p>Convert the human gmt file into the species of interest</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-1" tabindex="-1"></a><span class="fu">names</span>(originating_gmt<span class="sc">$</span>genesets) <span class="ot">&lt;-</span> originating_gmt<span class="sc">$</span>geneset.names</span>
<span id="cb74-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-2" tabindex="-1"></a></span>
<span id="cb74-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-3" tabindex="-1"></a>temp_genesets <span class="ot">&lt;-</span> <span class="fu">lapply</span>(originating_gmt<span class="sc">$</span>genesets,<span class="at">FUN=</span><span class="cf">function</span>(x){homologs<span class="sc">$</span>external_gene_name[<span class="fu">which</span>(homologs<span class="sc">$</span>hsapiens_homolog_associated_gene_name <span class="sc">%in%</span> <span class="fu">unlist</span>(x))]})</span>
<span id="cb74-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-4" tabindex="-1"></a></span>
<span id="cb74-5"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-5" tabindex="-1"></a>translated_genessets <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">geneset_names=</span>originating_gmt<span class="sc">$</span>geneset.names, </span>
<span id="cb74-6"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-6" tabindex="-1"></a>                             <span class="at">genesets =</span> temp_genesets,</span>
<span id="cb74-7"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb74-7" tabindex="-1"></a>                             <span class="at">geneset.descriptions=</span>originating_gmt<span class="sc">$</span>geneset.descriptions)</span></code></pre></div>
</div>
<div id="format-results-into-gmt-file-1" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Format results into GMT file<a href="translate-human-gmt-file-to-another-species-using-ensembl.html#format-results-into-gmt-file-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Convert lists to a tab delimited string of gene names</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb75-1" tabindex="-1"></a>translated_genessets<span class="sc">$</span>collapsed_genenames  <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(translated_genessets<span class="sc">$</span>genesets,</span>
<span id="cb75-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb75-2" tabindex="-1"></a>                                             <span class="at">FUN=</span><span class="cf">function</span>(x){</span>
<span id="cb75-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb75-3" tabindex="-1"></a>   <span class="fu">paste</span>(<span class="fu">unlist</span>(x),<span class="at">collapse =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb75-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb75-4" tabindex="-1"></a>}))</span></code></pre></div>
<p>The format of the GMT file is described <a href="here">https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:<em>Gene_Matrix_Transposed_file_format</em>.28.2A.gmt.29</a> and consists of rows with the following</p>
<ul>
<li>Name</li>
<li>Description</li>
<li>tab delimited list of genes a part of this geneset</li>
</ul>
<p>Write out the gmt file with genenames</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-1" tabindex="-1"></a>gmt_file_genenames <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name=</span>translated_genessets<span class="sc">$</span>geneset_names,</span>
<span id="cb76-2"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-2" tabindex="-1"></a>                                 <span class="at">description=</span>translated_genessets<span class="sc">$</span>geneset.descriptions,</span>
<span id="cb76-3"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-3" tabindex="-1"></a>                                 <span class="at">genes =</span> translated_genessets<span class="sc">$</span>collapsed_genenames)</span>
<span id="cb76-4"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-4" tabindex="-1"></a></span>
<span id="cb76-5"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-5" tabindex="-1"></a>gmt_genenames_filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(params<span class="sc">$</span>working_dir, <span class="fu">paste</span>(params<span class="sc">$</span>new_species,<span class="st">&quot;convertedfrom&quot;</span>,<span class="fu">basename</span>(dest_gmt_file),<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb76-6"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-6" tabindex="-1"></a></span>
<span id="cb76-7"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-7" tabindex="-1"></a><span class="fu">write.table</span>(<span class="at">x =</span> gmt_file_genenames,<span class="at">file =</span> gmt_genenames_filename,</span>
<span id="cb76-8"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-8" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb76-9"><a href="translate-human-gmt-file-to-another-species-using-ensembl.html#cb76-9" tabindex="-1"></a>            <span class="at">col.names=</span><span class="cn">TRUE</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="create-gmt-file-from-ensembl.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="create-enrichment-map-from-r-with-gprofiler-results.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/risserlin/CBW_pathways_workshop_R_notebooks/edit/main/05B-Convert_human_gmt_to_species_X.Rmd",
"text": "Suggest an edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cbw_workshop_notebooks.pdf", "https://github.com/risserlin/CBW_pathways_workshop_R_notebooks/raw/main/05B-Convert_human_gmt_to_species_X.Rmd"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
